src/gui/closeabletabwidget.cpp

```diff
@@ -0,0 +1,49 @@
+/*
+ * Bittorrent Client using Qt and libtorrent.
+ * Copyright (C) 2020  Will Da Silva <will@willdasilva.xyz>
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * In addition, as a special exception, the copyright holders give permission to
+ * link this program with the OpenSSL project's "OpenSSL" library (or with
+ * modified versions of it that use the same license as the "OpenSSL" library),
+ * and distribute the linked executables. You must obey the GNU General Public
+ * License in all respects for all of the code used other than "OpenSSL".  If you
+ * modify file(s), you may extend this exception to your version of the file(s),
+ * but you are not obligated to do so. If you do not wish to do so, delete this
+ * exception statement from your version.
+ */
+
+#include "closeabletabwidget.h"
+
+#include <QTabBar>
+#include <QMouseEvent>
+
+void MiddleClickTabBar::mouseReleaseEvent(QMouseEvent* event)
+{
+    const int tabIndex = tabAt(event->pos());
+    if ((event->button() == Qt::MiddleButton) && (tabIndex >= 0)) {
+        event->accept();
+        emit tabCloseRequested(tabIndex);
```

Comment:
I'm not sure emitting another signal is correct here. Perhaps you should just do `removeTab(tabIndex);`

Reply:
I don't recall why, but doing it that way was resulting in the removal of 2 tabs every time a single tab was removed.

Reply:
I tested `removeTab(tabIndex);` and it work flawless for me.

Reply:
Emitting `tabCloseRequested(tabIndex)` is the thing that clicking on close button does. So I would emulate its behavior instead of remove tabs directly.